src/include/platform/ConnectivityManager.h

```diff
@@ -171,6 +174,17 @@ class ConnectivityManager
     bool IsWiFiStationProvisioned();
     void ClearWiFiStationProvision();
     CHIP_ERROR GetAndLogWiFiStatsCounters();
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    CHIP_ERROR SetWiFiPAFAdvertisingEnabled(bool val);
+
+    typedef void (*OnConnectionCompleteFunct)(void * appState);
+    typedef void (*OnConnectionErrorFunct)(void * appState, CHIP_ERROR err);
+    CHIP_ERROR WiFiPAFConnect(const SetupDiscriminator & connDiscriminator, void * appState, OnConnectionCompleteFunct onSuccess,
```

Comment:
Again, this needs to have documentation and functionality for avoiding exploitable security bugs when appState is destroyed.

Reply:
We just pushed a commit which uses weak_ptr to check if the pointer is valid or not.